vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+used here exist from `{admiral}`, except for the `tte_sources` helper object,
+provided as an example from `{admiralonco}`.
+
+**Note**: *All examples assume CDISC SDTM and/or ADaM format as input unless
+otherwise specified.*
+
+## Required Packages
+
+The examples of this vignette require the following packages. (This is to be
+reviewed/updated once template code is written)
+
+```{r, warning=FALSE, message=FALSE}
+library(admiral)
+library(dplyr)
+library(admiral.test)
+library(lubridate)
+library(stringr)
+library(admiralonco)
+```
+
+# Programming Workflow
+
+* [Read in Data](#readdata)
+* [Merge TR with TU and derive new variables](#merge)
+* [Pre-processing of input records](#input)
+* [Derive Parameters (`TLDn`, `TLPn`)](#parameters)
+* [Derive Parameters (TSD) and ANL01FL](#tsd)
+* [Derive Baseline (BASETYPE, ABLFL, BASE)](#base)
+* [Derive NADIR](#nadir)
+* [Derive Change from Baseline (CHG, PCHG)](#chg)
+* [Derive Additional Flag Variables (e.g. CRFL, CRNFL, PDFL, PDFL](#addflag)
+* [Derive Additional Analysis Flags (e.g. ANL02FL, ANL03FL, ANL04FL](#analflag)
+* [Derive Analysis Sequence Number (`ASEQ`)](#aseq)
+* [Add ADSL Variables](#adslvars)
+
+## Read in Data {#readdata}
+
+To start, all data frames needed for the creation of the tumor results dataset
+should be read into the environment. This will be a company specific process.
+Some of the data frames needed may be `ADSL`, `ADRS` ,`RS`,  `TU` and `TR`.
+
+For example purpose, the SDTM and ADaM datasets (based on CDISC Pilot test
+data)---which are included in `{admiral.test}`---are used
+
+```{r}
+data("admiral_adsl")
+data("admiral_adrs")
+data("admiral_rs")
+data("admiral_tu")
+data("admiral_tr")
+adsl <- admiral_adsl
+adrs <- admiral_adrs
+tu <- admiral_tu
+tr <- admiral_tr
+rs <- admiral_rs
+
+tu <- convert_blanks_to_na(tu) %>%
+  filter(TUEVAL == "INVESTIGATOR")
+tr <- convert_blanks_to_na(tr) %>%
+  filter(TREVAL == "INVESTIGATOR" & TRGRPID == "TARGET" & TRTESTCD == "DIAMETER")
+rs <- convert_blanks_to_na(rs)
+```
+
+```{r echo=FALSE}
+# work-around for https://github.com/pharmaverse/admiral.test/issues/98
+# should be removed once the bug is fixed
+tr <- ungroup(tr)
+```
+
+```{r echo=FALSE}
+tr <- filter(tr, USUBJID %in% c("01-701-1015", "01-701-1023", "01-701-1028"))
+```
+
+At this step, it may be useful to join `ADSL` to your `TR` domain.  Only the
+`ADSL` variables used for derivations are selected at this step.  The rest of
+the relevant `ADSL` variables would be added later.
+
+```{r eval=TRUE}
+adsl_vars <- vars(RANDDT)
+tr <- derive_vars_merged(
+  tr,
+  dataset_add = adsl,
+  new_vars = adsl_vars,
+  by_vars = vars(STUDYID, USUBJID)
+)
+```
+
+## Merge TR with TU and derive new variables {#merge}
+
+Depending on your company specifications you may want to merge `TU` with `TR`.
+As an example, here we are keeping the tumor location, and deriving a new
+variable for the tumor group site. You may want to process other variables like
+`ADT`.
```

Comment:
Can you add the columns being kept and created in this text? i.e.:

Depending on your company specifications you may want to merge `TU` with `TR`.
As an example, here we are keeping the tumor location (**TULOC**), and deriving a new
variable for the tumor group site (**TULOCGR1**). You may want to process other variables like
`ADT`.

Reply:
done